EIPS/eip-6662.md

```diff
@@ -0,0 +1,116 @@
+
+```
+{
+    "title": "AuthenticationRequest",
+    "type": "object",
+    "properties": {
+        "entrypoint": {
+            "type": "string" | "$e2ee",
+            "description": "the entrypoint contract address",
+        },
+        "chainId": {
+            "type": "string" | "$e2ee",
+            "description": "the chain id",
+        },
+        "userOp": {
+            "type": "object" | "$e2ee",
+            "description": "UserOp defined by ERC-4337 without signature",
+        },
+        "encryptedData": {
+            "type": "string",
+            "description": "contains all encrypted fields"
+        },
+    }
+}
+```
+
+## Rationale
+
+To enable the new authentication workflow we described above, dApp needs to know two things:
+
+1. **Where is the authenticator?** This is solved by the **relayURI** field in struct **AuthenticationInfo**. Users can publish the uri as the account metadata which will be pulled by dApp to do service discovery.
+
+2. **What’s the format of AuthenticationRequest?** This is solved by the **schema** field in struct **AuthenticationInfo**. The schema defines the structure of the AuthenticationRequest object which is consumed by the authenticator. It can also be used to define extra fields for the relay service to enable flexible access control.
+
+### Relay Service Selection
+
+Each authenticator can provide a list of relay services. dApp should pull through the list of relay services in order to find the first workable one. All relay services under each authenticator must follow the same schema.
+
+### Signature Aggregation
+
+Multisig authentication could be enabled if multiple AuthenticatorInfos are provided under each smart contract account. Each authenticator can sign and submit signed user operations to bundler independently. These signatures will be aggregated by the Aggregator defined in ERC-4337.
+
+### Future Extension
+
+The **IAccountMetadata** interface could be extended per different requirements. For example, a new alias or avatar field could be defined for profile displaying.
+
+## Backwards Compatibility
+
+The new interface is fully backward compatible with ERC-4337.
+
+## Security Considerations
+
+### End to End Encryption
+
+To protect the user’s privacy and prevent front-running attacks, we should keep the data encrypted during transmission from dApps to authenticators. This could be done by adopting the JWE (JSON Web Encryption, RFC-7516) method. Before sending out AuthenticationRequest, we generate a symmetric CEK(Content Encryption Key) to encrypt fields with end to end encryption enabled, then encrypt the CEK with the signer's public key. dApp will pack the request into a JWE object and send it to the authenticator through the relay service. Relay service has no access to the end to end encrypted data since only the authenticator has the key to decrypt the CEK.
```

Comment:
`we generate a symmetric CEK`

Who is "we" here?

Reply:
I removed the "we" to use "CEK" as the subject.